### DISCUSSION TITLE
vcpkg from latest visual studio fails to find dependencies curl and libuv

### DISCUSSION THREAD

**Discussion by HallofFamer** [microsoft/vcpkg#41291] [General]
Hi, I've been able to get vcpkg to work with my software project in the past, but since the lastest update from Visual Studio it does not work anymore, instead gave out this error when I try to run the command with cmake(vcpkg.cmake is located in the D:/Software/Microsoft Visual Studio directory on my local machine): 

`cmake -DCMAKE_TOOLCHAIN_FILE:STRING="D:/Software/Microsoft Visual Studio/2022/Community/VC/vcpkg/scripts/buildsystems/vcpkg.cmake" -S . -B ./build
`

```
-- Running vcpkg install
Fetching registry information from https://github.com/microsoft/vcpkg (HEAD)...
curl does not exist
libuv does not exist
-- Running vcpkg install - failed
CMake Error at D:/Software/Microsoft Visual Studio/2022/Community/VC/vcpkg/scripts/buildsystems/vcpkg.cmake:899 (message):
  vcpkg install failed.  See logs for more information:
  C:\Users\Ordland\MyProject\build\vcpkg-manifest-install.log
Call Stack (most recent call first):
  D:/Software/Microsoft Visual Studio/2022/Community/Common7/IDE/CommonExtensions/Microsoft/CMake/CMake/share/cmake-3.29/Modules/CMakeDetermineSystem.cmake:146 (include)
  CMakeLists.txt:5 (project)
```

It seems rather strange to be as it was working for me just a couple of months ago, and now suddenly vcpkg is complaining that it cannot locate the dependencies `curl` and `libuv`. Below are the code in my vcpkg.json and vcpkg-configuration.json files, I've never modified them since a few months ago and they were working correctly before: 

vcpkg.json
```
{
  "name": "MyProject",
  "version": "2.0.0",
  "dependencies": [
    "curl", "libuv"
  ]
}
```

vcpkg-configuration.json
```
{
  "default-registry": {
    "kind": "git",
    "baseline": "638b1588be3a265a9c7ad5b212cef72a1cad336a",
    "repository": "https://github.com/microsoft/vcpkg"
  },
  "registries": [
    {
      "kind": "artifact",
      "location": "https://github.com/microsoft/vcpkg-ce-catalog/archive/refs/heads/main.zip",
      "name": "microsoft"
    }
  ]
}
```

Can someone plz help? I tried to google it and couldnt find any answers from reddit or stackoverflow, I dont believe I am the only one who has such an issue?

**Comment by HallofFamer** [comment]
It’s been almost a week, why is there no response so far? Do people even look at this discussion section at all? I also checked and there are many other questions not receiving a single answer either, why have such a discussion section when no one is there to answer questions anyway?

**Reply by dg0yt** [reply]
Community might wait for you to attach the logs.

**Comment by dg0yt** [comment]
Start with the printed instructions:

> See logs for more information:
>   C:\Users\Ordland\MyProject\build\vcpkg-manifest-install.log

Start with a clean build dir. `CMAKE_TOOLCHAIN_FILE` must be set on the first run.

**Reply by HallofFamer** [reply]
Yeah thats exactly what I did, though `CMAKE_TOOLCHAIN_FILE` was specified as a flag and it was working before. Also I did not attach the log file before since the information was not very useful, it basically repeated what the error message was on the terminal: 

```
Fetching registry information from https://github.com/microsoft/vcpkg (HEAD)...
curl does not exist
libuv does not exist
```

If only the log would be more useful I might have been able to find a solution myself, but the only thing it says is that curl and libuv do not exist. Again what I dont understand is why the same process was working a few months ago, and now suddenly it does not work anymore. If you have some suggestions why this could happen, even if just a hint rather than an actual solution, I will appreciate.

**Comment by xiaomao0o0o0o** [comment]
I have met the same problem when I use the iconv library, and I solved it after I replace "iconv" with the correct name of the library which is "libiconv" in the vcpkg.json file. Hope this will be useful to you .

**Comment by ColoredCarrot** [comment]
I had a similar symptom, with a different cause. For anyone who might be running into the same issue:

We have a custom registry hosted in an internal git repo. All the ports in that registry started failing with this message (`foo` does not exist).

The problem was that the git fetch of the registry was failing. Silently. The output was

```
Fetching registry information from ssh://git@example.com/foo-company/vcpkg.git (HEAD)...
foo does not exist
bar does not exist
...
```

However, when manually attempting `git clone ssh://git@example.com/foo-company/vcpkg.git`, there was finally an error message ("authenticity of host can't be established").

To make it even weirder: The fetch failure _wasn't_ silent when running vcpkg install (actually, cmake configure) manually. Then, it printed a helpful error like "error: failed to fetch ref HEAD from repository".
It was only failing silently when run as part of a PowerShell script. For some reason.
